### PR TITLE
Obsolete Extension Marketplace page

### PR DESCRIPTION
**Pull request by SBalslev**
The Extension Marketplace page is using an embeded view of Microsoft AppSource and the endpoint will be discontinued in 2024.
Marking the page as obsolete and that it will be replaced with a new Microsoft AppSource apps feature providing similar functionality.

Fixes [AB#493421](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/493421)

### CODE REVIEW

File: src/System Application/App/Extension Management/src/ExtensionMarketplace.Page.al [hunk 1]
```diff
@@ -19,6 +19,9 @@ page 2502 "Extension Marketplace"
     ApplicationArea = All;
     UsageCategory = Administration;
     Editable = false;
+    ObsoleteState = Pending;
```
Comment (qutreson) [inline src/System Application/App/Extension Management/src/ExtensionMarketplace.Page.al:23]: Shouldn't you introduce CLEAN tags and handle the references for this page? Or will you handle it in the next PR?

File: src/System Application/App/Extension Management/src/ExtensionMarketplace.Codeunit.al [hunk 2]
```diff
@@ -525,6 +525,12 @@ codeunit 2501 "Extension Marketplace"
     [EventSubscriber(ObjectType::Codeunit, Codeunit::"System Action Triggers", OpenAppSourceMarket, '', false, false)]
     local procedure OpenAppSourceMarket()
     begin
+#if not CLEAN24        
+#pragma warning disable AL0432
         Page.Run(Page::"Extension Marketplace");
+#pragma warning restore AL0432
+#else        
+        Hyperlink('https://appsource.microsoft.com/en-us/marketplace/apps?product=dynamics-365%3Bdynamics-365-business-central&page=1');
```
Comment (ChethanT) [inline src/System Application/App/Extension Management/src/ExtensionMarketplace.Codeunit.al]: shouldn't this be a forwardlink?

Comment (SBalslev) [inline src/System Application/App/Extension Management/src/ExtensionMarketplace.Codeunit.al]:
```suggestion
        Hyperlink('https://aka.ms/BCAppSource');
```

File: src/System Application/App/Extension Management/src/ExtensionManagement.Page.al [hunk 3]
```diff
@@ -259,7 +259,16 @@ page 2500 "Extension Management"
...
                     ToolTip = 'Browse the extension marketplace for new extensions to install.';
                     Visible = not IsOnPremDisplay;
+#if not CLEAN24                    
+#pragma warning disable AL0432
                     RunObject = page "Extension Marketplace";
+#pragma warning restore AL0432
+#else                    
+                    trigger OnAction()
+                    begin
+                        Hyperlink('https://appsource.microsoft.com/en-us/marketplace/apps?product=dynamics-365%3Bdynamics-365-business-central&page=1');
```
Comment (ChethanT) [inline src/System Application/App/Extension Management/src/ExtensionManagement.Page.al]: forwardlink?

**Review by sergiusignacius — DISMISSED** [review]
We could replace the appsource link with a fwlink just to be safe.

File: src/System Application/App/Extension Management/src/ExtensionManagement.Page.al (hunk 3 above)
Comment (SBalslev) [inline src/System Application/App/Extension Management/src/ExtensionManagement.Page.al]:
```suggestion
                        Hyperlink('https://aka.ms/bcappsource');
```

**Review by qutreson — APPROVED** [review]

**Review by thpeder — APPROVED** [review]

**Review by ChethanT — APPROVED** [review]

**Review by sergiusignacius — APPROVED** [review]

**Pull request merged**